eng/packages/http-client-csharp/generator/TestProjects/Local/Basic-TypeSpec/src/Generated/BasicTypeSpecClient.cs

```diff
@@ -52,7 +51,7 @@ public BasicTypeSpecClient(Uri endpoint, ApiKeyCredential keyCredential, BasicTy
 
             _endpoint = endpoint;
             _keyCredential = keyCredential;
-            Pipeline = HttpPipelineBuilder.Build(options, new HttpPipelinePolicy[] {  });
+            Pipeline = HttpPipelineBuilder.Build(options, new HttpPipelinePolicy[] { new Azure.Core.AzureKeyCredentialPolicy(_keyCredential, AuthorizationHeader) });
```

Comment:
I'm curious why it is a fully qualified name?

Reply:
I think this is due to `AzureKeyCredentialPolicy` is from Azure.Core shared files, not a part of public Azure.Core package.

Reply:
I don't see why it coming from shared source should matter. Can you file an issue in the MGC repo for this?

Reply:
But from a user perspective, they would expect the full namespace to not be used when it is not needed. So we still need to come up with a way to avoid this, so I suggest filing an issue to MGC.

Reply:
https://github.com/microsoft/typespec/issues/5104 created
was just trying to explain why it's not working

Reply:
Thanks! The explanation will be helpful context when trying to come up with a solution.